routers/api/v1/repo/issue_label.go

```diff
@@ -173,14 +177,42 @@ func DeleteIssueLabel(ctx *context.APIContext) {
 		return
 	}
 
-	label, err := models.GetLabelByID(ctx.ParamsInt64(":id"))
-	if err != nil {
-		if models.IsErrLabelNotExist(err) {
+	mode := ctx.Query("mode")
+	var (
+		label *models.Label
+	)
+	if mode == "id" {
+		label, err = models.GetLabelByID(ctx.ParamsInt64(":id"))
+		if err != nil {
+			if models.IsErrLabelNotExist(err) {
+				ctx.Error(http.StatusUnprocessableEntity, "", err)
+				return
+			}
+			ctx.Error(http.StatusInternalServerError, "parseLabel", err)
+			return
+		}
+	} else if mode == "name" {
+		label, err = models.GetLabelInRepoByName(ctx.Repo.Repository.ID, ctx.Params(":id"))
+		if err != nil {
+			if models.IsErrLabelNotExist(err) {
+				ctx.Error(http.StatusUnprocessableEntity, "", err)
+				return
+			}
+			ctx.Error(http.StatusInternalServerError, "parseLabel", err)
+			return
+		}
+	} else {
+		exist := false
+		label, exist, err = models.GetLabelByIDOrName(ctx.Params(":id"), ctx.Repo.Repository.ID)
+		if err != nil {
+			ctx.Error(http.StatusInternalServerError, "GetLabelByIDOrName", err)
+			return
+		}
+
+		if !exist {
```

Comment:
Use a `switch` block and default to id not mixed by default - otherwise this a breaking change.